code/modules/unit_tests/create_and_destroy.dm

```diff
@@ -138,6 +138,27 @@ GLOBAL_VAR_INIT(running_create_and_destroy, FALSE)
 	//Now that we've qdel'd everything, let's sleep until the gc has processed all the shit we care about
 	var/time_needed = SSgarbage.collection_timeout[GC_QUEUE_CHECK]
 	var/start_time = world.time
+	// spin until the first item in the filter queue is older than start_time
+	var/filter_queue_finished = FALSE
+	var/list/filter_queue = SSgarbage.queues[GC_QUEUE_FILTER]
+	while(!filter_queue_finished)
+		if(!length(filter_queue))
+			filter_queue_finished = TRUE
+			break
+		var/list/oldest_item = filter_queue[1]
+		//Pull out the time we deld at
+		var/qdel_time = oldest_item[1]
+		if(qdel_time > start_time) // Everything we care about has graduated to the check queue now!
+			filter_queue_finished = TRUE
+			break
+		if(world.time > start_time + GC_FILTER_QUEUE + 2 MINUTES) // This is much faster than the check queue.
```

Comment:
would prefer to use the variable from ssgarbage rather then the define, since it covers more cases